File: bubble sort/BubbleSort.java
import java.util.Scanner;

class BubbleSort {
	static void Sort(int[] arr) {
		for (int unsort = arr.length-1; unsort > 0; unsort--) {
			for (int i = 0; i < unsort; i++) {
				if (arr[i] > arr[i+1]) {
					int temp = arr[i];
					arr[i] = arr[i+1];
					arr[i+1] = temp;
				}
			}
		}
	}

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int n = input.nextInt();
		int[] numbers = new int[n];

		for (int i = 0; i < n; i++) {
			numbers[i] = input.nextInt();
		}

		Sort(numbers);

		for (int i = 0; i < n; i++) {
			System.out.print(i == n-1 ? numbers[i] : numbers[i] + " ");
		}
	}
}

File: bubble sort/bubble_sort.c
#include <stdio.h>

void Sort(int *arr, int *size);

int main()
{
	int n = 0;
	scanf("%d", &n);
	int numbers[n];

	for (int i = 0; i < n; i++) {
		scanf("%d", &numbers[i]);
	}

	Sort(numbers, &n);

	for (int i = 0; i < n; i++) {
		i == n-1 ? printf("%d", numbers[i]) : printf("%d ", numbers[i]);
	}
	return 0;
}

void Sort(int *arr, int *size) {
	for (int unsort = *size-1; unsort > 0; unsort--) {
		for (int i = 0; i < unsort; i++) {
			if (arr[i] > arr[i+1]) {
				int temp = arr[i];
				arr[i] = arr[i+1];
				arr[i+1] = temp;
			}
		}
	}
}
